aws/biganimal-preflight-aws

```diff
@@ -0,0 +1,478 @@
+#!/usr/bin/env bash
+#
+# Copyright 2022 EnterpriseDB Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used as a preflight check to tell if the given region
+# in your AWS account can meet the requirement for BigAnimal to create the
+# PostgreSQL cluster in.
+#
+# Given one the below input:
+#   - (mandatory) AWS account ID
+#   - (mandatory) region
+#   - PostgreSQL cluster's instance type
+#   - whether the HA (High Availability) is used for the PostgreSQL cluster
+#   - network type (private, or public)
+# it checks the below requirement:
+#   - the AWS CLI is configured with access to the specified account ID
+#   - the given region:
+#     * if there is enough vCPU quota left for this PostgreSQL cluster type
+#       in your AWS Account
+#     * if there is enough NLB (Network Load Balancer) quota left to expose the service
+#       for you to access the PostgreSQL cluster
+#
+# The output of this script tells any unsatisfied condition and report in the form
+# of table.
+#
+# For more details, please refer to:
+#  https://www.enterprisedb.com/docs/biganimal/latest/getting_started/01_check_resource_limits
+#
+set -e
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+[ "${BASH_VERSINFO:-0}" -lt 4 ] && echo -e "${RED}This script does not support Bash version 3 and below${NC}"
```

Comment:
why don't we exit the script?
```
[ "${BASH_VERSINFO:-0}" -lt 4 ] && echo -e "${RED}This script does not support Bash version 3 and below${NC}"
```

Reply:
we probably can remove this line as we didn't use any bash 4+ feature. This is just a copy-paste from the biganimal-preflight-azure script I wrote.

Reply:
I'm using an associative array here https://github.com/EnterpriseDB/cloud-utilities/pull/42/files#diff-ca92d0630a0520e3150028111d3a3d94596ddec22288b63f23506eea8db93711R464 that is supported from Bash 4... maybe exiting the script is a good change that I should do. Thanks!